src/frontend/src/features/home/components/PersonalizeMeetingDialog.tsx

```diff
@@ -0,0 +1,134 @@
+import { FormEvent, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Button, Dialog, type DialogProps, Text } from '@/primitives'
+import { HStack } from '@/styled-system/jsx'
+import { RiSpam2Fill } from '@remixicon/react'
+import { navigateTo } from '@/navigation/navigateTo.ts'
+import { useCreateRoom } from '@/features/rooms'
+import {
+  FieldError,
+  Form as RACForm,
+  Input,
+  Label,
+  TextField,
+} from 'react-aria-components'
+import { css } from '@/styled-system/css'
+
+export const PersonalizeMeetingDialog = ({
+  isOpen,
+  ...dialogProps
+}: Omit<DialogProps, 'title'>) => {
+  const { t } = useTranslation('home', {
+    keyPrefix: 'personalizeMeetingDialog',
+  })
+  const { mutateAsync: createRoom } = useCreateRoom()
+  const [roomSlug, setRoomSlug] = useState('')
+  const [serverErrors, setServerErrors] = useState<Array<string>>([])
+
+  const onSubmit = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+    createRoom({ slug: roomSlug })
+      .then((data) =>
+        navigateTo('room', data.slug, {
+          state: { create: true, initialRoomData: data },
+        })
+      )
+      .catch((e) => {
+        const msg =
+          e.statusCode === 400
+            ? t('errors.server.taken')
+            : t('errors.server.unknown')
+        setServerErrors([msg])
+      })
+  }
+
+  const validationErrors = []
+  if (roomSlug.length < 5) {
+    validationErrors.push(t('errors.validation.length'))
+  }
+  if (!roomSlug.match(/^[a-z0-9]+$/)) {
+    validationErrors.push(t('errors.validation.spaceOrSpecialCharacter'))
+  }
```

Comment:
I like how simple this approach is were most of devs wants to use libraries to validate their forms :)